pkg/kubelet/kubelet.go

```diff
@@ -1499,8 +1531,20 @@ func (kl *Kubelet) GetPodStatus(podFullName string, uid types.UID) (api.PodStatu
 
 	info, err := dockertools.GetDockerPodInfo(kl.dockerClient, spec, podFullName, uid)
 
+	for _, c := range spec.Containers {
+		containerStatus := info[c.Name]
+		containerStatus.Ready = kl.readiness.IsReady(containerStatus)
+		info[c.Name] = containerStatus
+	}
+
 	var podStatus api.PodStatus
 	podStatus.Phase = getPhase(&spec, info)
+	if isPodReady(&spec, info) {
```

Comment:
Should there be a not-ready condition or should the ready condition just be omitted from the pod's Conditions slice? Is the not ready condition a ConditionKind of Ready and a ConditionStatus of None or does it need a separate ConditionKind?